.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,50 @@
+name: Build with emscripten
+
+on: push
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install cached emscripten
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: 'emsdk'
+          key: 'emsdk'
+
+      - name: Install emscripten
+        if: steps.cache.outputs.cache-hit != 'true'
+        run: |
+          curl -L https://github.com/emscripten-core/emsdk/archive/HEAD.tar.gz | tar xz
```

Comment:
Please use a URL that has a release version number.

Reply:
Done. Pinned to v3.1.30